src/sentry/api/serializers/models/user.py

```diff
@@ -250,22 +264,24 @@ def serialize(
     ) -> DetailedUserSerializerResponse:
         d = cast(DetailedUserSerializerResponse, super().serialize(obj, attrs, user))
 
-        # XXX(dcramer): we don't use is_active_superuser here as we simply
-        # want to tell the UI that we're an authenticated superuser, and
-        # for requests that require an *active* session, they should prompt
-        # on-demand. This ensures things like links to the Sentry admin can
-        # still easily be rendered.
-        d["isSuperuser"] = obj.is_superuser
-        d["permissions"] = [up.permission for up in attrs["permissions"]]
-        d["authenticators"] = [
-            {
-                "id": str(a.id),
-                "type": a.interface.interface_id,
-                "name": str(a.interface.name),
-                "dateCreated": a.created_at,
-                "dateUsed": a.last_used_at,
-            }
-            for a in attrs["authenticators"]
-        ]
-        d["canReset2fa"] = attrs["canReset2fa"]
+        # safety check to never return this informatoin if the acting user is not 1) this user, 2) an admin
+        if user.id == obj.id or user.is_superuser:
+            # XXX(dcramer): we don't use is_active_superuser here as we simply
+            # want to tell the UI that we're an authenticated superuser, and
+            # for requests that require an *active* session, they should prompt
+            # on-demand. This ensures things like links to the Sentry admin can
+            # still easily be rendered.
+            d["isSuperuser"] = obj.is_superuser
+            d["permissions"] = sorted(attrs["permissions"])
+            d["authenticators"] = [
+                {
+                    "id": str(a.id),
+                    "type": a.interface.interface_id,
+                    "name": str(a.interface.name),
+                    "dateCreated": a.created_at,
+                    "dateUsed": a.last_used_at,
+                }
+                for a in attrs["authenticators"]
+            ]
+            d["canReset2fa"] = attrs["canReset2fa"]
```

Comment:
who can tell me what this is entirely used for? it was the only one i wasnt certain was admin-only

Reply:
so basically if a user is only a member of 1 org, we allow owners/managers of the org to reset the user's 2fa

Reply:
ah ok so still need exposed :+1: will fix